test/stdlib/Result.swift

```diff
@@ -0,0 +1,155 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import Swift
+
+let ResultTests = TestSuite("Result")
+
+fileprivate enum Err: Error, Equatable {
+  case err
+  case derr
+}
+
+fileprivate let string = "string"
+
+ResultTests.test("Construction") {
+  let result1: Result<String, String> = .success(string)
+  let result2: Result<String, String> = .failure(string)
+  let string1: String? = {
+    switch result1 {
+      case let .success(string): 
+        return string
+      case .failure: 
+        return nil
+    }
+  }()
+  let string2: String? = {
+    switch result2 {
+      case let .failure(string): 
+        return string
+      case .success: 
+        return nil
+    }
+  }()
+
+  expectEqual(string1, string2)
```

Comment:
This test could succeed by `.success` constructing a failure and `.failure` constructing a success, making `string1` and `string2` both `nil`. It'd be better to have two separate `expectEqual(_:_:)` lines, each comparing one of the strings to the constant.

Reply:
Added suggested comparison to constant strings and made the opposite cases unreachable.